tests/subsys/dsp/basicmath/src/f16.c

```diff
@@ -30,6 +30,21 @@ static void test_zdsp_add_f16(
 	/* Run test function */
 	zdsp_add_f16((float16_t *)input1, (float16_t *)input2, output, length);
 
+	/* Validate output */
+	zassert_true(
+		test_snr_error_f16(length, output, (float16_t *)ref,
+			SNR_ERROR_THRESH),
+		ASSERT_MSG_SNR_LIMIT_EXCEED);
+
+	zassert_true(
+		test_rel_error_f16(length, output, (float16_t *)ref,
+			REL_ERROR_THRESH),
+		ASSERT_MSG_REL_ERROR_LIMIT_EXCEED);
+
+	/* Validate in-place processing */
+	memcpy(output, (float16_t *)input1, length * sizeof(float16_t));
+	zdsp_add_f16(output, (float16_t *)input2, output, length);
```

Comment:
Please make this a separate test (i.e. `test_zdsp_add_f16_in_place`) since this is testing a special type of operation.